commands/hugo_test.go

```diff
@@ -0,0 +1,314 @@
+package commands
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/hugo/helpers"
+	"github.com/spf13/hugo/hugofs"
+	"github.com/spf13/hugo/hugolib"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+)
+
+func clearConfigFile() {
+	viper.SetConfigFile("")
+	cfgFile = ""
+	destination = ""
+	source = ""
+	baseURL = ""
+}
+
+func TestUserErrorCommandNewUserError(t *testing.T) {
+	viper.Reset()
+	defer viper.Reset()
+	expectedUserError := commandError{"test\n", true}
+	actualUserError := newUserError("test")
+	assert.Equal(t, expectedUserError, actualUserError)
+}
+
+func TestSystemErrorCommandNewSystemError(t *testing.T) {
+	viper.Reset()
+	defer viper.Reset()
+	expectedSystemError := commandError{"test\n", false}
+	actualSystemError := newSystemError("test")
+	assert.Equal(t, expectedSystemError, actualSystemError)
+}
+
+func TestUserErrorCommandNewFormattedUserError(t *testing.T) {
+	viper.Reset()
+	defer viper.Reset()
+	expectedUserError := commandError{"test:12", true}
+	actualUserError := newUserErrorF("test:%d", 12)
+	assert.Equal(t, expectedUserError, actualUserError)
+}
+
+func TestSystemErrorCommandNewFormattedSystemError(t *testing.T) {
+	viper.Reset()
+	defer viper.Reset()
+	expectedSystemError := commandError{"test:12", false}
+	actualSystemError := newSystemErrorF("test:%d", 12)
+	assert.Equal(t, expectedSystemError, actualSystemError)
+	if actualSystemError != expectedSystemError {
+		t.Errorf("Actual (%v) returned command did not equal expected (%v) command.", actualSystemError, expectedSystemError)
+	}
+}
+
+func TestUserErrorReturnError(t *testing.T) {
+	viper.Reset()
+	defer viper.Reset()
+	cError := commandError{"test error", false}
+	expectedError := "test error"
+	actualError := cError.Error()
+	assert.Equal(t, expectedError, actualError)
+}
+
+func TestIsUserError(t *testing.T) {
+	viper.Reset()
+	defer viper.Reset()
+	var errors = []struct {
+		err error
```

Comment:
I would add expected as a bool to this struct, then add that to the list below. Then you can add tests for errors that should not be flagged as user errors. Also include userError and serverError in the test table.